### PR TITLE
Search: use alias to link to search results of subprojects

### PR DESCRIPTION
**Pull request by stsewd**
Currently we are showing `(result from {subproject_slug})`
instead of the chosen alias.
With this now we can show the proper alias. The alias is tracked in the VersionData namedtuple that is passed into the context.

Also, start tracking the doctype of the version (it changes when the
version changes so we are good).
With this we can have the correct urls for all projects
https://github.com/readthedocs/readthedocs.org/blob/1675fd3ea55947ef1de681c87df625ff5bc43abd/readthedocs/search/serializers.py#L113

### CODE REVIEW

File: readthedocs/search/documents.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ class PageDocument(RTDDocTypeMixin, Document):
     # Metadata
     project = fields.KeywordField(attr='project.slug')
     version = fields.KeywordField(attr='version.slug')
+    doctype = fields.KeywordField(attr='version.documentation_type')
```
Comment (ericholscher) [inline readthedocs/search/documents.py:72]: Does this require reindexing?

**Review by ericholscher — COMMENTED** [review]
This seems like a good approach. I dunno about VersionData having `project_alias` in it though. That feels messy and confusing, but I guess its fine for now.

File: readthedocs/search/documents.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/search/documents.py:72]: yeah, but we can do it the next year, we aren't using it yet (still querying from the db or falling back to None). Tracking it here https://github.com/readthedocs/readthedocs.org/issues/7762.

**Comment by stsewd** [comment]
> I dunno about VersionData having project_alias in it though

Yeah, I was thinking about having a structure like `project: {'alias': 'alias', version: {...}}`.

**Comment by stsewd** [comment]
I have changed it to use another structure.

**Review by ericholscher — APPROVED** [review]
This looks good 👍

**Pull request merged**